jax/_src/random.py

```diff
@@ -1254,13 +1286,16 @@ def categorical(key, logits, axis=-1, shape=None):
   if shape is None:
     shape = batch_shape
   else:
+    shape = tuple(shape)
```

Comment:
All of the other functions in this file support shapes of any sequence type.  `categorical` only supported `tuple`.   This cast repairs that.